include/bcs/EconomouDielectricBC.h

```diff
@@ -0,0 +1,86 @@
+//* This file is part of Zapdos, an open-source
+//* application for the simulation of plasmas
+//* https://github.com/shannon-lab/zapdos
+//*
+//* Zapdos is powered by the MOOSE Framework
+//* https://www.mooseframework.org
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef EconomouDielectricBC_H
+#define EconomouDielectricBC_H
+
+#include "IntegratedBC.h"
+
+class EconomouDielectricBC;
+
+template <>
+InputParameters validParams<EconomouDielectricBC>();
+
+/**
+ * Implements a simple constant Neumann BC where grad(u)=value on the boundary.
+ * Uses the term produced from integrating the diffusion operator by parts.
+ */
+class EconomouDielectricBC : public IntegratedBC
+{
+public:
+  /**
+   * Factory constructor, takes parameters so that all derived classes can be built using the same
+   * constructor.
+   */
+  EconomouDielectricBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual() override;
+  virtual Real computeQpJacobian();
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar);
+
+  /// Value of grad(u) on the boundary.
```

Comment:
I think this is another leftover comment from NeumannBC?

Reply:
Removed.